File: src/scoring/getSimplifiedDiscardScore.ts
import { Card } from "../interfaces/card.interface";
import { scoreFifteens } from "./scoreFifteens";
import { scorePairs } from "./scorePairs";

export const getSimplifiedDiscardScore = (discard: Card[]): number => {
  if (discard.length !== 2) {
    throw new Error("Discard must contain exactly two cards.");
  }

  let score = 0;
  score += scoreFifteens(discard);
  score += scorePairs(discard);

  return score;
};

File: crib_calc/scoring/get_simplified_discard_score.py
from crib_calc.card import Card
from crib_calc.scoring.score_fifteens import score_fifteens
from crib_calc.scoring.score_pairs import score_pairs


def get_simplified_discard_score(discard: list[Card]) -> int:
    if len(discard) != 2:
        raise ValueError("Discard must contain exactly two cards.")
    return score_fifteens(discard) + score_pairs(discard)
